pytest/tools/prober/prober.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+
+"""
+Prober that is compatible with cloudprober.
+Prober verifies that any block since the genesis can be retrieved from a node.
+Makes 3 separate RPC requests: Get genesis height, get head, get a block.
+If the block contains chunks, then make one more RPC request to get a chunk.
+
+Run like this:
+./prober --url http://my.test.net.node:3030
+
+"""
+
+import argparse
+import datetime
+import json
+import pathlib
+import random
+import requests
+import sys
+import time
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+
+import configured_logger
+
+logger = configured_logger.new_logger("stderr", stderr = True)
+
+def json_rpc(method, params, url):
+    try:
+        j = {
+            'method': method,
+            'params': params,
+            'id': 'dontcare',
+            'jsonrpc': '2.0'
+        }
+        start_time = time.time()
+        r = requests.post(url, json=j, timeout=5)
+        latency_ms = (time.time() - start_time)
+        print(f'prober_request_latency_ms{{method="{method}",url="{url}"}} {latency_ms:.2f}')
```

Comment:
is this for cloudprober?

Reply:
This metric is re-exported by cloudprober and it's available in Grafana.